airflow/example_dags/example_xcom.py

```diff
@@ -18,50 +18,76 @@
 
 """Example DAG demonstrating the usage of XComs."""
 from airflow import DAG
-from airflow.operators.python import PythonOperator
+from airflow.decorators import task
+from airflow.operators.python import PythonOperator, get_current_context
 from airflow.utils.dates import days_ago
 
 value_1 = [1, 2, 3]
 value_2 = {'a': 'b'}
+value_3 = 'some value'
 
 
 def push(**kwargs):
     """Pushes an XCom without a specific target"""
     kwargs['ti'].xcom_push(key='value from pusher 1', value=value_1)
 
 
+@task
+def push_with_taskflow(val=value_2):
+    """
+    Pushes an XCom from within a task-decorated function aka using the Taskflow API.
+
+    .. note::
+        Retrieving the current execution context using the ``get_current_context()`` method should only be
+        done when using the Taskflow API.
+    """
+
+    context = get_current_context()
+    context['ti'].xcom_push(key='other value from pusher 1', value=val)
```

Comment:
No, this is wrong, and I think this much simpler way will work:

```suggestion
@task
def push_with_taskflow(ti, val=value_2):
    """
    Pushes an XCom from within a task-decorated function aka using the Taskflow API.
    """
    ti.xcom_push(key='other value from pusher 1', value=val)
```

(But could you test it please)

Reply:
i.e. I'm not sure _when_ `get_current_context()` is ever actually needed.

Reply:
> No, this is wrong, and I think this much simpler way will work:
> 
> (But could you test it please)

Tested this out. Just needed to make `ti` an optional arg -- otherwise DAG parsing would throw an exception -- and it worked just fine.  Seems to make the use of `get_current_context()` more nebulous.

Reply:
To close the gap on this comment, I've gone ahead and removed any TaskFlow API-related changes (reference [this comment](https://github.com/apache/airflow/pull/16866#discussion_r689944981)).  We can discuss later when I tackle another example DAG issue but passing in the `ti` object as optional works just fine and I'll incorporate this change then.